lib/internal/modules/cjs/loader.js

```diff
@@ -806,6 +807,7 @@ Module.prototype._compile = function(content, filename) {
   var exports = this.exports;
   var thisValue = exports;
   var module = this;
+  if (requireDepth === 0) statCache = new Map();
```

Comment:
So your fix here is to clear the stat cache for each top-level require?

Is this the historical intention, for stats to always be live at top-level require time?

Reply:
Yes, that was the original implementation. I extended the cache a bit and it turned out that this broke the repl in this case.

Reply:
Would it make sense to explicitly clear the stat cache then during the requireDepth decrement itself, when it is set to zero? May make it clearer what is going on and avoid the null checks?

Reply:
We could do that but it actually adds two checks that are not only run once per loaded module but once per require call. So I would rather keep it here. The `null` checks also have to stay in place. We have some stat calls before the first require (I did not dig into the reason).